tests/manage/monitoring/conftest.py

```diff
@@ -431,3 +442,421 @@ def corrupt_pg():
+    #  - pool is missing (likely a product bug)
+    # either way, the fixture can't continue ...
+    assert ceph_pool is not None, f"pool {ceph_pool_name} should exist"
+    # ... to compute PVC size (values in bytes)
+    total = ceph_pool["stats"]["max_avail"] + ceph_total_stored
+    max_avail_gi = ceph_pool['stats']['max_avail'] / 2**30
+    logger.info(f"MAX AVAIL of {ceph_pool_name} is {max_avail_gi} Gi")
+    target = total * target_percentage
+    to_utilize = target - ceph_total_stored
+    pvc_size = round(to_utilize / 2**30)  # GiB
+    logger.info((
+        f"fixture is going to request {pvc_size} Gi volume "
+        f"to reach {target/2**30} Gi of total cluster utilization, which "
+        f"is {target_percentage*100}% of the total capacity"))
+    return pvc_size
+
+
+def fio_to_dict(fio_output):
+    """"
+    Parse fio output and provide parsed dict it as a result.
+    """
+    fio_output_lines = fio_output.splitlines()
+    for line_num, line in enumerate(fio_output_lines):
+        if line == "{":
+            break
+        else:
+            logger.info(line)
+    fio_parseable_output = "\n".join(fio_output_lines[line_num:])
+    fio_report = yaml.safe_load(fio_parseable_output)
+    return fio_report
+
+
+def workload_fio_storageutilization(
+        fixture_name,
+        target_percentage,
+        project,
+        fio_pvc_dict,
+        fio_job_dict,
+        fio_configmap_dict,
+        measurement_dir,
+        tmp_path,
+):
+    """
+    This function implements core functionality of fio storage utilization
+    workload fixture. This is necessary because we can't parametrize single
+    general fixture over multiple parameters (it would mess with test case id
+    and polarion test case tracking).
+    """
+    if fixture_name.endswith("rbd"):
+        storage_class_name = "ocs-storagecluster-ceph-rbd"
+        ceph_pool_name = "ocs-storagecluster-cephblockpool"
+    elif fixture_name.endswith("cephfs"):
+        storage_class_name = "ocs-storagecluster-cephfs"
+        ceph_pool_name = "ocs-storagecluster-cephfilesystem-data0"
+    else:
+        raise Exception("unexpected volume type, ocs-ci code is wrong")
+
+    # make sure we communicate what is going to happen
+    logger.info((
+        f"starting {fixture_name} fixture, "
+        f"using {storage_class_name} storage class "
+        f"backed by {ceph_pool_name} ceph pool"))
+
+    pvc_size = get_storageutilization_size(target_percentage, ceph_pool_name)
+
+    # For cephfs we can't use fill_fs because of BZ 1763808 (the process
+    # will get *Disk quota exceeded* error instead of *No space left on
+    # device* error).
+    # On the other hand, we can't use size={pvc_size} for rbd, as we can't
+    # write pvc_size bytes to a filesystem on a block device of {pvc_size}
+    # size (obviously, some space is used by filesystem metadata).
+    if fixture_name.endswith("rbd"):
+        fio_conf = textwrap.dedent("""
```

Comment:
consider saving these 2 conf under ocs_ci/templates/workloads dir

Reply:
This is not a good idea. The fio config file is a core of the workload, and should be directly part of the fixture, so that the intentions of the workload are clearly visible (esp. when I have slightly different different configurations for cephfs vs xfs/rbd volume).

Reply:
But that will not be flexible to be used by other functions in the project

Reply:
This fio workload could be reused in the form of this workload fixture. Moreover there is no way to reuse such fio config files via functionality available in ocs-ci module, neither is any test case doing this.